### PR TITLE
Hello world

### PR DESCRIPTION
**Pull request by miklosduma**

### CODE REVIEW

File: docs/tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,208 @@
...
+    >>> sparkl('connect', SSE_URL)
+    {...'tag': 'pong'}
+    >>> sparkl('login', USER, PASSWORD)
+    {'attr': {'administrator': False, ...'name': ...}, 'tag': 'user'}
+
+We log into SPARKL using our username and password. If you don't have a registered user yet, use the ``register=True`` flag with the ``login`` command.
+
+    Remember, if you're running SSE locally, you can connect to it through ``http://localhost:8000``.
+
+ .. code-block:: python
```
Comment (lucafavatellawork) [inline docs/tutorial.rst]: Is this leading space on purpose? At this point it looks like the code block is quoted in the markup rendering by GitHub.

File: docs/tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,208 @@
...
+
+Since the ``prefix`` and ``name`` fields already have their values we do not need to set them again.
+
+We call the request operation the same way as we would a solicit. It returns the very same result, except in this case it's not a response but a reply.
+
+A solicit has a response to it, a request has a reply to it. So what's the actual difference?
+
+The main difference is that a solicit operation triggers a transaction that will end when all the transaction goals - such as a response to the solicit - are satisfied.
+
+The purpose of request/reply operations is to help satisfy these goals.
```
Comment (lucafavatellawork) [inline docs/tutorial.rst]: Questions that I have reading this (great!) tutorial (documenting is out of scope for this PR):

Q1. Can user put a mix with multiple requests with (1) same request fields, (2) same reply name and (3) same reply fields?

Q2. Can a request send/write to outside the request? (Side effects.)

Q3. Can a request read/receive from outside the request? (Determinism.)

Q4. Can user put a mix with multiple requests that have same request fields? In this case, could the backend determine to call a request though realize it does not lead to reply needed for response hence backtrack and attempt distinct request? Could side effects have happened that cannot be rolled back?

Q5. Can a user put a mix with multiple requests, with same reply name and same reply fields? Dually to Q4: what about backtracking and side effects?

File: docs/tutorial.rst (hunk 1 above)
Comment (miklosduma) [reply docs/tutorial.rst]: Yes, it is supposed to be a side-note. But no, it should not be rendered together with the code-block. Maybe an extra line will solve this. Thanks!

File: docs/tutorial.rst (hunk 2 above)
Comment (miklosduma) [reply docs/tutorial.rst]: - XML nodes with the same name are not permitted as far as I know. For the name, the full path counts. E.g. you can have folder1/request and folder2/request but you cannot have folder1/request and folder1/request. So yes, and no, depending on how the requests are organised into folders/mixes :)
- The invocation of a request might have a side effect on the service that implements it, so yes, there can be side effects - and those might affect other operations on the same service too
- Yes, the request can read e.g. state from the service that implements it
- A request will only be invoked if the fields sent by its reply contribute to the response (or another transaction goal such as a consume operation). If there are two or more requests, each of which can contribute to the same goal, only one of them will be called. I think it is random which one. So there will be no need to rollback. When the mix is written, the possible execution paths are calculated. At execution time, one of the paths will be picked. I.e. no rollback needed :)
- See above, only one of the requests will be actually used at a time :) 

Thanks for the questions, I'll try to cover some of these in the next tutorials

File: docs/tutorial.rst (hunk 1 above)
Comment (miklosduma) [reply docs/tutorial.rst]: Nope, there was an extra space there, now should be fixed, thanks!

**Pull request merged**